### ISSUE TITLE
Focus lands twice on input element when role is set to combobox along with aria-label or placeholder in virtual PC cursor mode

### ISSUE THREAD

**Issue by Sneha174** [FreedomScientific/standards-support#189]
### Summary

> When an input element has its role set to combobox along with certain other attributes like placeholder or aria-label, focus lands twice on the input element in virtual PC cursor mode
> 
> Example:
> 1. Here is a simple html to reproduce the issue:
<pre><code>
&lt;html&gt;
    &lt;body&gt;
        &lt;div&gt;Correct behaviour - role:none and aria-label is set&lt;/div&gt;
        &lt;input role="none" aria-label="Test"  /&gt;
        &lt;br /&gt;
        &lt;div&gt;Wrong behaviour - role:combobox and aria-label is set&lt;/div&gt;
        &lt;input role="combobox" aria-label="Test"  /&gt;
        &lt;br /&gt;
        &lt;div&gt;Correct behaviour - role:none and placeholder is set&lt;/div&gt;
        &lt;input role="none" placeholder="test" /&gt;
        &lt;br /&gt;
        &lt;div&gt;Wrong behaviour - role:combobox and placeholder is set&lt;/div&gt;
        &lt;input role="combobox" placeholder="test" /&gt;
        &lt;br /&gt;
        &lt;div&gt;Sample repro ends&lt;/div&gt;
    &lt;/body&gt;
&lt;/html&gt;
</code></pre>
> 2. Enable virtual PC cursor and navigate this with up/down arrows.

### Expected result

> When role="combobox" the focus should not land twice on the input element on pressing up/down arrow

### Actual result

> Focus lands twice on the input element when role="combobox"

### Example

> A simple html to reproduce this issue:
<pre><code>
&lt;html&gt;
    &lt;body&gt;
        &lt;div&gt;Correct behaviour - role:none and aria-label is set&lt;/div&gt;
        &lt;input role="none" aria-label="Test"  /&gt;
        &lt;br /&gt;
        &lt;div&gt;Wrong behaviour - role:combobox and aria-label is set&lt;/div&gt;
        &lt;input role="combobox" aria-label="Test"  /&gt;
        &lt;br /&gt;
        &lt;div&gt;Correct behaviour - role:none and placeholder is set&lt;/div&gt;
        &lt;input role="none" placeholder="test" /&gt;
        &lt;br /&gt;
        &lt;div&gt;Wrong behaviour - role:combobox and placeholder is set&lt;/div&gt;
        &lt;input role="combobox" placeholder="test" /&gt;
        &lt;br /&gt;
        &lt;div&gt;Sample repro ends&lt;/div&gt;
    &lt;/body&gt;
&lt;/html&gt;
</code></pre>

#### JAWS version and build number
JAWS version 2019.1901.66 ILM

 
#### Operating System and version
Windows 10

#### Browser and version:
Chrome Version 73.0.3683.86 (Official Build) (64-bit)

**Comment by menovak** [comment]
hi Sneha174,

I follow what you are doing, but I would ask you what you would expect JAWS to do with the following code snippet?

`<div> Visble label versus aria-label.</div> `
`<p></p>`
`<label for="test2"> Label combo </label>`
`<input role="combobox" id="test2">`
`<p></p>`
`<div>What is the difference in Virtual PC navigation mode?</div>`
`<p></p>`
`<input  aria-label="Label combo" role="combobox" >`

Testing using Chrome 75, JAWS 2019.1904, by browsing (arrow leys, Virtual PC Curson On), I believe these announce identical and have same number of key presses to navigate through.

Regards,
Mark, TPG Engineer
`

**Comment by JAWS-test** [comment]
It's not a bug. For most form elements, the label is output first and the element is output in the next reading step. This is independent of the browser and JAWS version. There is only a small inconsistency because, for example, with checkboxes, the label is output in the same reading step and not extra before it. I don't know if there is a reason for this. If not, this could be standardized.

**Comment by JAWS-test** [comment]
If it becomes unified, the output of the checkbox should become the default. In fact, there is a problem with the output of form field labels if the label for is in the source code not directly in front of the form field (which in my opinion neither the HTML specification nor the WCAG requires). In this case, the form fields (except checkboxes) are output with a wrong or without a label during linear reading. For example:
```
<label for="test1">edit 1</label>
<label for="test2">edit 2</label>
<input id="test1">
<input id="test2">
```

**Comment by JAWS-test** [comment]
Additionally, it is confusing that the output is different depending on whether label for or aria-labelledby is used. With aria-labelledby a remote label is repeated at the form field, with label for not (except for checkboxes). The output can be checked with the following code:

```
<table role=presentation>
	<tr>
		<td><span id="test1">edit 1</span>
		<td><span id="test2">check 2</span>
		<td><label for="test3">edit 3</label>
		<td><label for="test4">check 4</label>
	</tr>
	<tr>
		<td><input aria-labelledby="test1">
		<td><input aria-labelledby="test2" type=checkbox>
		<td><input id="test3">
		<td><input id="test4" type=checkbox>
	</tr>
</table>
```
